doc/manual/src/introduction.md

```diff
@@ -1,193 +1,144 @@
+This is a _binary deployment model_.
 
-## Functional package language
+[Hydra](https://hydra.nixos.org/), the Nix build cluster, has been producing binary artifacts and making them available in the [public cache](http://cache.nixos.org/) for many years.
+The cache allows running software going back as far as 2010, especially where sources cannot be obtained otherwise any more.
 
-Packages are built from _Nix expressions_, which is a simple
-functional language.  A Nix expression describes everything that goes
-into a package build task (a “derivation”): other packages, sources,
-the build script, environment variables for the build script, etc.
-Nix tries very hard to ensure that Nix expressions are
-_deterministic_: building a Nix expression twice should yield the same
-result.
+Nix also makes it easy to set up your own binary cache.
 
-Because it’s a functional language, it’s easy to support
-building variants of a package: turn the Nix expression into a
-function and call it any number of times with the appropriate
-arguments.  Due to the hashing scheme, variants don’t conflict with
-each other in the Nix store.
+## Distributed builds
 
-## Transparent source/binary deployment
+Nix can **transparently delegate tasks to remote machines**, for example to build on different architectures or operating systems, or run many builds in parallel.
+These machines can share the same cache.
+
+Setting up your own remote build machine only requires installing Nix (or NixOS), registering SSH keys, and configuring the client when to use it.
+
+## Functional configuration language
+
+Which artifacts Nix produces is traditionally specified with _Nix expressions_.
+
+A Nix expression describes everything that goes into a “derivation”:
+the executable to run, input files and environment variables it can read, and command line arguments.
+
+Nix tries very hard to ensure that **Nix expressions are deterministic**:
+both evaluating the Nix expression and running an executable configured that way  should yield the same result every time.
+
+Because **it’s a purely functional programming language**, it allows expressing highly complex compositions and their variants succinctly.
+Since unique files have unique file system paths, variants won’t conflict with each other in the Nix store.
+
+## Nixpkgs software distribution
+
+[Nixpkgs](https://github.com/NixOS/nixpkgs) has become [the largest, most up-to-date free software repository in the world](https://repology.org/repositories/graphs).
+It holds tens of thousands of working packages **built from source** and **instantly available in binary form**.
+It offers mechanisms to **compose or customise software** according to your needs.
+
+## NixOS Linux distribution
+
+[NixOS](https://github.com/NixOS/nixpkgs/tree/master/nixos) is a Linux distribution based on Nix and Nixpkgs, and allows you to **manage the entire system configuration declaratively** through the Nix language.
```

Comment:
I got the intention intention that people don't immediately get what declarative means here. Maybe some variant of this makes it more clear. 
```suggestion
[NixOS](https://github.com/NixOS/nixpkgs/tree/master/nixos) is a Linux distribution based on Nix and Nixpkgs, and allows you to **manage the entire system configuration** through the Nix language in a central configuration file.
```